prober/dns.go

```diff
@@ -246,7 +251,14 @@ func ProbeDNS(ctx context.Context, target string, module config.Module, registry
 	level.Info(logger).Log("msg", "Making DNS query", "target", targetIP, "dial_protocol", dialProtocol, "query", module.DNS.QueryName, "type", qt, "class", qc)
 	timeoutDeadline, _ := ctx.Deadline()
 	client.Timeout = time.Until(timeoutDeadline)
+	// we don't use the rtt value returned fom client.Exchange because that
+	// includes only the time to exchange messages with the server _after_
+	// the connection is created. We would in principle have three phases:
+	// resolve (probe_dns_lookup_time_seconds), connect (request - rtt),
```

Comment:
To keep things similar to the metric in the other probes, we should break this out into its phases.

Reply:
OK, will do

Reply:
I added the phase label and added the three time measurements, as indicated in the original comment.